### ISSUE TITLE
Problems running Terra code

### ISSUE THREAD

**Issue by kapecp** [luapower/terra#3]
### Hello

I have problems running Terra code using LuaPower on OSX 10.11.6 on a MacBook Pro (Retina, 15-inch, Mid 2014) with NVIDIA GeForce GT 750M card, with CUDA drivers installed

I have tested 2 approaches

1. using prebuild LuaPower from luapower.com
2. built LuaJIT and Terra from source

### problems with approach 1)

following work OK:
```
./luajit test.lua
```

where _test.lua_ contains only simple Lua code

however I can't start any Terra code:

running:
`./luajit -lterra`
- result: OK

running:
`./luajit test.lua`
- _test.lua_ contains only Lua code, but requires Terra (same as `-lterra` flag) 
- result: runs Lua code, but finishes with segmentation fault

running:
`./luajit test.t`
- _test.t_ contains simple Terra code: `var a = 1`
- result: 
```
luapower-all-master/bin/osx64/luajit-bin: test.t:4: '=' expected near 'a'
[1]    44145 segmentation fault  ./luajit -lterra test.t

```
Maybe the segmentation fault problem is due to that the prebuild binaries are for OSX 10.13,
but what causes the parsing problem?  

### problems with approach 2)

I compiled LuaJIT, Terra, LLVM from LuaPower sources, I did not modify any build scripts

LuaJIT works OK for Lua code, but running (_test.t_ contains `var a = 1`):
`./luajit test.t` or just `./luajit -lterra`
results in this error:
```
.//bin/osx64/luajit: ./terralib.lua:740: failed to initialize target for LLVM Triple: nvptx64-nvidia-cuda (No available targets are compatible with this triple.)
```
I am not familiar with CUDA development, so I do not know how to fix this problem (as it is deep in the complex terralib.lua), or whether this is a LuaPower or Terra problem (having functional CUDA would be nice)

**Comment by capr** [comment]
Hi, 

Looks like there's a problem with the OSX Terra binaries. I'll look into it when I'll get some time, thanks for reporting it. As for CUDA, I didn't include nor test CUDA at all with this build of terra because even though I do happen to have a NVIDIA card, with luapower I'm focused more on cross-platform solutions so I can't prioritize this.

**Comment by capr** [comment]
... wait, do you want to have CUDA working or is this just an error that you want to get rid of to make terra work irrespective of CUDA? I think I got it backwards... will look into it, though my Mac is a little behind on OSX 10.13.

**Comment by kapecp** [comment]
CUDA support would be nice, but currently I (and my team) don't need it
and thanks for your time ;-)

**Comment by capr** [comment]
I've rebuilt terra with llvm 6.0.1 on all platforms (llvm binaries are also available with the luapower/llvm package -- see the releases section on github). Couple of issues:

* I didn't have XCode on the machine see where system includes are, so you have to set the system includes dir manually on `terra_luapower.lua`.
* CUDA wasn't built in LLVM so it's not in Terra either (the CUDA backend is unmaintained in LLVM, CUDA 9 is broken on Windows and it requires an NVIDIA card ... hardly a cross-platform solution, so kinda out-of-scope for luapower anyway, I'm not sure I'll ever include it).
* 16 tests fail consistently on all platforms and they seem to be all related to saveobj() arg passing (I'll look more into it when I'll have the time).
* binaries need OSX >= 10.12 ... I should probably make this the minimum accepted OSX version in luapower overall (currently it's 10.7).

**Comment by capr** [comment]
I just noticed that you have OSX 10.11, damn. I decided on 10.12 as a minimum version yesterday by looking at the OSX market share, but I'll probably revert back to 10.7. I'll do a rebuild tonight. I have no idea what that min-version flag means in practical terms anyway.

**Comment by capr** [comment]
Updated LLVM and Terra binaries to OSX 10.7 min version. Let me know if you still have problems so I can close this.

**Comment by kapecp** [comment]
The new build seems to work OK, thanks!
notes:

- the CUDA problem might be related to how LLVM is built: You use `-DLLVM_TARGETS_TO_BUILD=X86`, but for CUDA it should be `-DLLVM_TARGETS_TO_BUILD="X86;NVPTX"` - I will try to rebuild with these flags...

**Comment by capr** [comment]
Yes, like I mentioned earlier, CUDA is left out of the build and I commented out that like that tries to initialize the triple in terralib so that it doesn't crash anymore.
